gateway/src/shard/builder.rs

```diff
@@ -14,6 +14,7 @@ use twilight_model::gateway::{payload::update_status::UpdateStatusInfo, Intents}
 ///
 /// [`ShardBuilder::large_threshold`]: struct.ShardBuilder.html#method.large_threshold
 #[derive(Debug)]
+#[non_exhaustive]
```

Comment:
i don't think there can ever be a 3rd type of error here? too few and too many already covers all posiblities